pandas/_libs/reduction.pyx

```diff
@@ -177,6 +177,8 @@ cdef class _BaseGrouper:
             object.__setattr__(cached_ityp, '_index_data', islider.buf)
             cached_ityp._engine.clear_mapping()
             object.__setattr__(cached_typ._data._block, 'values', vslider.buf)
+            object.__setattr__(cached_typ._data._block, 'mgr_locs',
+                               slice(len(vslider.buf)))
```

Comment:
im a little surprised this worked, since mgr_locs is a property (and the setter casts the slice to BlockPlacement), i thought youd have to go with the explicit `cached_typ._data.block.mgr_locs = slice(len(vslider.buf))`

Reply:
So does this prevent this from throwing a ValueError in the first place? Maybe this is an interim state, but confused on why this is required along with the changes to groupby.ops

Reply:
I'm pretty sure this prevents the ValueError, so the changed exception catching should be reverted

Reply:
Sorry, fixed.